File: backend/src/main/java/com/NBE3_4_2_Team4/domain/asset/main/validation/annotation/ValidAssetHistoryReq.java
package com.NBE3_4_2_Team4.domain.asset.main.validation.annotation;

import com.NBE3_4_2_Team4.domain.asset.main.validation.validator.AssetHistoryReqValidator;
import jakarta.validation.Constraint;
import jakarta.validation.Payload;

import java.lang.annotation.*;

@Documented
@Constraint(validatedBy = AssetHistoryReqValidator.class)
@Target({ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
public @interface ValidAssetHistoryReq {
    String message() default "Invalid date range";
    Class<?>[] groups() default {};
    Class<? extends Payload>[] payload() default {};
}

File: backend/src/main/java/com/NBE3_4_2_Team4/domain/asset/main/validation/annotation/ValidAssetHistoryReq.kt
package com.NBE3_4_2_Team4.domain.asset.main.validation.annotation

import com.NBE3_4_2_Team4.domain.asset.main.validation.validator.AssetHistoryReqValidator
import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass

@MustBeDocumented
@Constraint(validatedBy = [AssetHistoryReqValidator::class])
@Target(
    AnnotationTarget.CLASS
)
@Retention(AnnotationRetention.RUNTIME)
annotation class ValidAssetHistoryReq(
    val message: String = "Invalid date range",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = []
)

